### ISSUE TITLE
Issue with updating state of hocked component. 

### ISSUE THREAD

**Issue by rublemangoo** [Shalimov/react-hooks-combine#31]
withStateHandlers update state in wrong way, and could reset value of oner field in state. 
Example:

```javascript
withStateHandlers(() => {
			const { search, page } = queryString.parse(location.search)

			return {
				a: 1,
                                 b: 2
			}
		}, {
			setA: () => data => ({ data }),
			setB: () => data => ({ data }),
		}),
```
If  you will call setA, field b will be erroneously updated.

**Comment by Shalimov** [comment]
Thx, to raise it up.
withStateHandlers right now is working wrong with state ref inside, gonna fix it as soon as

**Comment by Shalimov** [comment]
Fixed in version 1.3.0 (it's already available on NPM) but be aware of some API changes there
